synapse/storage/data_stores/main/end_to_end_keys.py

```diff
@@ -332,6 +336,204 @@ def get_e2e_cross_signing_key(self, user_id, key_type, from_user_id=None):
             from_user_id,
         )
 
+    @cached(num_args=1)
+    def _get_bare_e2e_cross_signing_keys(self, user_id):
+        """Dummy function.  Only used to make a cache for
+        _get_bare_e2e_cross_signing_keys_bulk.
+        """
+        raise NotImplementedError()
+
+    @cachedList(
+        cached_method_name="_get_bare_e2e_cross_signing_keys",
+        list_name="user_ids",
+        num_args=1,
+    )
+    def _get_bare_e2e_cross_signing_keys_bulk(self, user_ids: list) -> dict:
```

Comment:
```suggestion
    def _get_bare_e2e_cross_signing_keys_bulk(self, user_ids: List[str]) -> Dict[str, Dict[str, dict]]:
```

(You might need to import from `typing`)